apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -310,6 +386,45 @@ Gallery.prototype = {
     }.bind(this));
   },
 
+  switchToGalleryFrame: function() {
+    var galleryFrame = this.client.findElement('iframe[src*="' +
+                                          Gallery.ORIGIN + '"]');
+    this.waitFor(galleryFrame);
```

Comment:
Now I'm just confused. If you just used findElement() to get the galleryFrame, what does it mean to wait for it? Does that wait for it to be visible?  Below you use waitFor() on a selector, not an element. Can you just do that here?

Probably I just dont' understand marionette well enough to get what is going on here.